tests/test_standard_tags.py

```diff
@@ -160,24 +160,51 @@ def test_route(self):
         interfaces.library.add_span_tag_validation(request=r, tags=tags)
 
 
-@released(
-    dotnet="2.13.0",
-    golang="1.39.0",
-    java="0.107.1",
-    nodejs="3.2.0",
-    php="0.76.0",
-    python=PYTHON_RELEASE_GA_1_1,
-    ruby="?",
-)
+@released(dotnet="?", golang="?", java="?")
+@released(nodejs="?", php="?", python="?", ruby="?")
 @coverage.basic
 class Test_StandardTagsClientIp(BaseTestCase):
     """Tests to verify that libraries annotate spans with correct http.client_ip tags"""
 
-    def test_client_ip(self):
+    @classmethod
+    def setup_class(cls):
+        """Send two_request, on with an attack, another one without attack"""
+        get = cls().weblog_get
+
         headers = {"X-Cluster-Client-IP": "10.42.42.42, 43.43.43.43, fe80::1"}
-        r = self.weblog_get("/waf/", headers=headers)
+        attack_headers = {"User-Agent": "Arachni/v1"}
+        cls.request_with_attack = get("/waf/", headers=headers | attack_headers)
+        cls.request_without_attack = get("/waf/", headers=headers)
 
-        tags = {
-            "http.client_ip": "43.43.43.43",
-        }
-        interfaces.library.add_span_tag_validation(request=r, tags=tags)
+    def test_client_ip(self):
+        """Test http.client_ip is always reported in the default scenario which has ASM enabled"""
+
+        def validator(span):
+            meta = span.get("meta", {})
+            assert "http.client_ip" in meta, "missing http.client_ip tag"
+
+            got = meta["http.client_ip"]
+            expected = "43.43.43.43"
+            assert got == expected, f"unexpected http.client_ip value {got} instead of {expected}"
+
+            return True
+
+        interfaces.library.add_span_validation(request=self.request_with_attack, validator=validator)
+        interfaces.library.add_span_validation(request=self.request_without_attack, validator=validator)
+
+    def test_client_ip_with_appsec_event(self):
+        """Test that meta tag are correctly filled when an appsec event is present and ASM is enabled"""
+
+        def validator(span):
+            meta = span.get("meta", {})
+
+            # ASM should report extra IP-related span tags.
+            assert "appsec.event" in meta, "missing appsec.event tag"
+            assert "network.client.ip" in meta, "missing network.client.ip tag"
```

Comment:
Should the conditional `network.client.ip` go away now that there is `http.client_ip`?

Context: I was wondering about removing it since it's currently inaccurate in Ruby (not possible to reliably get the TCP address that connected, so it [does some resolution](https://github.com/rack/rack/blob/ab9cd7416c1b5253cd280d75704bafd87510c120/lib/rack/request.rb#L414-L433) not entirely unlike `http.client_ip` but also not as complete)

Alternatively, instead of `Rack::Request#ip` I could switch to use Rack's `env['REMOTE_ADDR']`, which may or may not be present, and should represent the reverse proxy, if any.

Reply:
Feel free to iterate on this when you implement this for Ruby.
I was also wondering whether I should test `network.client.ip` or not.
Before removing it, we should check its RFC again to see why did we define this tag.